dao/elasticsearch/filesystem.go

```diff
@@ -335,3 +342,45 @@ func (dao *ControlPlaneDao) ReadyDFS(serviceID string, _ *int) (err error) {
 	err = dao.facade.ResetLock(ctx, serviceID)
 	return
 }
+
+// TagSnapshot tags an existing snapshot with 1 or more strings
+func (dao *ControlPlaneDao) TagSnapshot(request model.TagSnapshotRequest, newTagList *[]string) error {
+	ctx := datastore.Get()
+
+	// synchronize the dfs
```

Comment:
Why do you think that we need to lock the DFS if we are adding a tag to an existing snapshot?